File: src/main/java/github/cweijan/ultimate/util/Log.java
package github.cweijan.ultimate.util;

import kotlin.jvm.JvmStatic;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Log {

    private static Logger loggerChecker = LoggerFactory.getLogger(Log.class);

    public static Logger getLogger(){

        return getLoggerInner();
    }

    private static Logger getLoggerInner() {
        StackTraceElement stackTrace = (new Throwable()).getStackTrace()[2];
        return LoggerFactory.getLogger(stackTrace.getClassName());
    }

    @JvmStatic
    public static void error(Object content) {
        if (loggerChecker.isErrorEnabled()) {
            getLoggerInner().error(content != null ? content.toString() : null, "");
        }

    }

    @JvmStatic
    public static void error(Object content, Throwable throwable) {
        if (loggerChecker.isErrorEnabled()) {
            getLoggerInner().error(content != null ? content.toString() : null, "", throwable);
        }

    }

    @JvmStatic
    public static void warn(Object content) {
        if (loggerChecker.isWarnEnabled()) {
            getLoggerInner().warn(content != null ? content.toString() : null, "");
        }

    }

    @JvmStatic
    public static void debug(Object content) {
        if (loggerChecker.isDebugEnabled()) {
            getLoggerInner().debug(content != null ? content.toString() : null, "");
        }

    }

    @JvmStatic
    public static void info(Object content) {
        if (loggerChecker.isInfoEnabled()) {
            getLoggerInner().info(content != null ? content.toString() : null, "");
        }

    }

    private Log() {
    }
}

File: src/main/java/github/cweijan/ultimate/util/Log.kt
package github.cweijan.ultimate.util

import org.slf4j.Logger
import org.slf4j.LoggerFactory

object Log {

    private val loggerChecker = LoggerFactory.getLogger(Log::class.java)

    private val logger: Logger
        get() {
            val stackTrace = Throwable().stackTrace[2]
            val invokeClassName = stackTrace.className
            return LoggerFactory.getLogger(invokeClassName)
        }

    @JvmStatic
    fun error(content: Any?) {

        if (loggerChecker.isErrorEnabled) logger.error(content?.toString() + "")
    }

    @JvmStatic
    fun error(content: Any?, throwable: Throwable) {

        if (loggerChecker.isErrorEnabled) logger.error(content?.toString() + "", throwable)
    }

    @JvmStatic
    fun warn(content: Any?) {

        if (loggerChecker.isWarnEnabled) logger.warn(content?.toString() + "")
    }

    @JvmStatic
    fun debug(content: Any?) {

        if (loggerChecker.isDebugEnabled) logger.debug(content.toString() + "")

    }

    @JvmStatic
    fun info(content: Any?) {

        if (loggerChecker.isInfoEnabled) logger.info(content?.toString() + "")

    }

}